modules/button/react/lib/IconButton.tsx

```diff
@@ -129,45 +122,85 @@ export const IconButtonCon = styled('button', {
-            ...getAccentSelector(colors.blueberry500),
+            ...(toggled ? focusRing(2, 2) : {}),
           },
-          ...getBackgroundSelector(colors.frenchVanilla100),
+          '&:hover': {
+            backgroundColor: colors.blueberry500,
+          },
+          backgroundColor: colors.blueberry400,
+          borderColor: colors.blueberry400,
           ...getFillSelector(colors.frenchVanilla100),
-          ...getAccentSelector(colors.blueberry400),
+          ...getAccentSelector(colors.frenchVanilla100),
         };
       case IconButtonVariant.Inverse:
         return {
-          '&:hover span .wd-icon-fill, span .wd-icon-fill': {
-            fill: colors.frenchVanilla100,
+          '&:focus&:hover, &:focus, &:active': {
+            backgroundColor: colors.frenchVanilla100,
+            ...getFillSelector(colors.licorice500),
+            ...getAccentSelector(colors.licorice500),
+          },
+          '&:not([disabled]):focus': {
+            backgroundColor: colors.frenchVanilla100,
+            ...(toggled
+              ? focusRing(2, 2, true, false, 'currentColor', colors.frenchVanilla100)
+              : {}),
+          },
+          '&:hover': {
+            backgroundColor: colors.frenchVanilla100,
+            ...getFillSelector(colors.licorice500),
+            ...getAccentSelector(colors.licorice500),
           },
-          ...getBackgroundSelector(colors.frenchVanilla100),
-          ...getAccentSelector(colors.licorice200),
+          backgroundColor: colors.blueberry400,
+          borderColor: colors.blueberry400,
+          ...getFillSelector(colors.frenchVanilla100),
+          ...getAccentSelector(colors.frenchVanilla100),
         };
       case IconButtonVariant.InverseFilled:
         return {
           backgroundColor: 'rgba(0, 0, 0, 0.4)',
-          '&:focus': {
-            backgroundColor: 'rgba(0, 0, 0, 0.4)',
+          '&:focus&:hover, &:focus, &:active': {
+            backgroundColor: colors.frenchVanilla100,
+            ...getFillSelector(colors.licorice500),
+            ...getAccentSelector(colors.licorice500),
           },
-          ...getBackgroundSelector(colors.frenchVanilla100),
-          ...getFillSelector(colors.frenchVanilla100),
-          ...getAccentSelector(colors.licorice200),
-          '&:active': {
-            ...getAccentSelector(colors.licorice200),
+          '&:not([disabled]):focus': {
+            backgroundColor: colors.frenchVanilla100,
+            ...(toggled
+              ? focusRing(2, 2, true, false, 'currentColor', colors.frenchVanilla100)
+              : {}),
+          },
+          '&:hover': {
+            backgroundColor: colors.frenchVanilla100,
+            ...getFillSelector(colors.licorice500),
+            ...getAccentSelector(colors.licorice500),
           },
+          borderColor: colors.blueberry400,
+          ...getFillSelector(colors.frenchVanilla100),
+          ...getAccentSelector(colors.frenchVanilla100),
         };
```

Comment:
I think this can be cleaned up for sure, but I finally got it to a reasonable state where they somewhat work, but wanted to get feedback on this before cleaning it up a bit.